histomicstk/preprocessing/color_deconvolution/utils.py

```diff
@@ -0,0 +1,35 @@
+import numpy
+
+from .stain_color_map import stain_color_map
+
+def get_stain_vector(args, index):
+    """Get the stain corresponding to args.stain_$index and
+    args.stain_$index_vector.  If the former is not "custom", the
+    latter must be None.
+
+    """
+    args = vars(args)
+    stain = args['stain_' + str(index)]
+    stain_vector = args['stain_' + str(index) + '_vector']
+    if stain == 'custom':
+        if stain_vector is None:
+            raise ValueError('If "custom" is chosen for a stain, '
+                             'a stain vector must be provided.')
+        return stain_vector
+    else:
+        if stain_vector is None:
+            return stain_color_map[stain]
+        raise ValueError('Unless "custom" is chosen for a stain, '
+                         'no stain vector may be provided.')
+
+def get_stain_matrix(args, count=3):
+    """Get the stain matrix corresponding to the args.stain_$index and
+    args.stain_$index_vector arguments for values of index 1 to count.
+    Return a numpy array of column vectors.
+
+    """
+    return numpy.array([get_stain_vector(args, i+1) for i in range(count)]).T
+
+__all__ = (
+    'get_stain_vector',
+)
```

Comment:
Unfortunately it's more complicated. Relative imports only work in packages, something which the scattered executable files in `/server` are not. Cf. http://stackoverflow.com/a/73149 and comments.

I can think of a couple ways to try dealing with it, but they all require some reorganization.
- Create a secondary package alongside `histomicstk` -- `histomicstk_server` or some such. It would also have to be installed or otherwise on the `PYTHON_PATH` for `from histomicstk_server import utils` or the like to work.
- Organize the various CLIs into a package. This will require adding `__init__.py`s in several folders, and invoking the CLIs with `python -m path.to.CLI.CLI` instead of `python /path/to/CLI/CLI.py` (some variation of which is how I presume the CLIs are being immediately invoked currently). I'm not sure what all else would have to be changed for such a new launching scheme to work.